packages/valory/skills/task_execution/models.py

```diff
@@ -88,7 +88,9 @@ def __init__(self, *args: Any, **kwargs: Any) -> None:
         self.max_block_window: int = self._ensure_get("max_block_window", kwargs, int)
         # maps the request id to the number of times it has timed out
         self.request_id_to_num_timeouts: Dict[int, int] = defaultdict(lambda: 0)
-        mech_to_config_dict: Dict[str, Dict[str, bool]] = kwargs.get("mech_to_config")
+        mech_to_config_dict: Dict[str, Dict[str, bool]] = kwargs.get(
+            "mech_to_config", None
+        )
```

Comment:
Isn't it better to fail early?

Reply:
Isn't the default behaviour of the get() to return empty obj in case of not existent key ? I mean, in both cases if mech_to_config doesn't exist will be None.

Reply:
Added due to linter checks, can revert back

Reply:
No, actually the default raises a `KeyError`.

But even in this case, the typing is wrong as it should be converted to `Optional`. What was the check you were trying to resolve?

Reply:
But even in this case, the typing is wrong as it should be converted to `Optional`. What was the check you were trying to resolve?

Reply:
error: Incompatible types in assignment (expression has type "Optional[Any]", variable has type "Dict[str, Dict[str, bool]]")  [assignment]

Reply:
Yes, but it's still optional after the change you performed. So let's keep it as it was and open an issue since it's unrelated to the PR.